include/nbl/video/IGPUCommandPool.h

```diff
@@ -39,24 +39,27 @@ class IGPUCommandPool : public core::IReferenceCounted, public IBackendObject
         friend class CCommandSegment;
 
     public:
-        virtual ~ICommand() {}
+        virtual ~ICommand() = default;
 
         // static void* operator new(std::size_t size) = delete;
         static void* operator new[](std::size_t size) = delete;
         // static void* operator new(std::size_t size, std::align_val_t al) = delete;
         static void* operator new[](std::size_t size, std::align_val_t al) = delete;
 
-        // static void operator delete  (void* ptr) = delete;
         static void operator delete[](void* ptr) = delete;
-        static void operator delete  (void* ptr, std::align_val_t al) = delete;
         static void operator delete[](void* ptr, std::align_val_t al) = delete;
-        static void operator delete  (void* ptr, std::size_t sz) = delete;
         static void operator delete[](void* ptr, std::size_t sz) = delete;
-        static void operator delete  (void* ptr, std::size_t sz, std::align_val_t al) = delete;
         static void operator delete[](void* ptr, std::size_t sz, std::align_val_t al) = delete;
 
         inline uint32_t getSize() const { return m_size; }
 
+        void operator delete(ICommand* ptr, std::destroying_delete_t) { ptr->~ICommand(); }
+        void operator delete( ICommand* ptr, std::destroying_delete_t,
+                                std::align_val_t al ) { ptr->~ICommand(); }
+        void operator delete( ICommand* ptr, std::destroying_delete_t, std::size_t sz )  { ptr->~ICommand(); }
+        void operator delete( ICommand* ptr, std::destroying_delete_t,
+                                std::size_t sz, std::align_val_t al ) { ptr->~ICommand(); }
```

Comment:
what happens if you make them protected?

Reply:
You can do that.

Reply:
ok then lets make it protected and @achalpandeyy will merge if your changes work